### PR TITLE
resnet at Dygraph mode

### PR DESCRIPTION
**Pull request by junjun315**
add resnet at dygraph。with README

### CODE REVIEW

File: fluid/dygraph/resnet/train.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        y = self.conv(inputs)
+        y = self.pool2d_max(y)
+        for bottleneck_block in self.bottleneck_block_list:
+            y = bottleneck_block(y)
+        y = self.pool2d_avg(y)
+        y = self.out(y)
+        return y
+
+
+class DygraphResnet():
```
Comment (velconia) [inline fluid/dygraph/resnet/train.py]: object?

File: fluid/dygraph/resnet/train.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    def train(self):
+        batch_size = train_parameters["batch_size"]
+        batch_num = 10000
```
Comment (velconia) [inline fluid/dygraph/resnet/train.py]: remove this

File: fluid/dygraph/resnet/train.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        with fluid.dygraph.guard():
+
+            resnet = ResNet("resnet")
+            optimizer = optimizer_setting(train_parameters)
+            train_reader = paddle.batch(
+                paddle.dataset.flowers.train(use_xmap=False),
```
Comment (velconia) [inline fluid/dygraph/resnet/train.py]: remove this

File: fluid/dygraph/resnet/train.py [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+                batch_size=batch_size)
+
+            dy_param_init_value = {}
```
Comment (velconia) [inline fluid/dygraph/resnet/train.py]: remove this

File: fluid/dygraph/resnet/train.py [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+            for param in resnet.parameters():
+                dy_param_init_value[param.name] = param.numpy()
+
```
Comment (velconia) [inline fluid/dygraph/resnet/train.py]: remove this....

File: fluid/dygraph/resnet/train.py [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+            for eop in range(epoch):
+                for batch_id, data in enumerate(train_reader()):
+                    if batch_id >= batch_num:
```
Comment (velconia) [inline fluid/dygraph/resnet/train.py]: remove..

File: fluid/dygraph/resnet/train.py [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+                    loss = fluid.layers.cross_entropy(input=out, label=label)
+                    avg_loss = fluid.layers.mean(x=loss)
+
+                    dy_out = avg_loss.numpy()
+                    avg_loss.backward()
+
+                    optimizer.minimize(avg_loss)
+                    resnet.clear_gradients()
+
+                    print(eop, batch_id, dy_out)
```
Comment (velconia) [inline fluid/dygraph/resnet/train.py]: Add more debug info

File: fluid/dygraph/resnet/train.py (hunk 1 above)
Comment (junjun315) [reply fluid/dygraph/resnet/train.py]: deal

File: fluid/dygraph/resnet/train.py (hunk 2 above)
Comment (junjun315) [reply fluid/dygraph/resnet/train.py]: deal

File: fluid/dygraph/resnet/train.py (hunk 3 above)
Comment (junjun315) [reply fluid/dygraph/resnet/train.py]: deal

File: fluid/dygraph/resnet/train.py (hunk 7 above)
Comment (junjun315) [reply fluid/dygraph/resnet/train.py]: deal

File: fluid/dygraph/resnet/train.py (hunk 6 above)
Comment (junjun315) [reply fluid/dygraph/resnet/train.py]: deal

File: fluid/dygraph/resnet/train.py (hunk 5 above)
Comment (junjun315) [reply fluid/dygraph/resnet/train.py]: deal

File: fluid/dygraph/resnet/train.py (hunk 4 above)
Comment (junjun315) [reply fluid/dygraph/resnet/train.py]: deal

**Review by velconia — APPROVED** [review]
LGTM

**Pull request merged**